Src/DependencyCollector/Microsoft.AI.DependencyCollector/DependencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,189 @@
+// ------------------------------------------------------------
+
+namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Reflection;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.DependencyCollector.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.Extensions.DiagnosticAdapter;
+
+    /// <summary>
+    /// Diagnostic listener implementation that listens for events specific to outgoing depedency requests.
+    /// </summary>
+    public class DependencyCollectorDiagnosticListener : IObserver<DiagnosticListener>
+    {
+        /// <summary>
+        /// Add Application Insights Dependency Collector services to this .NET Core application.
+        /// </summary>
+        /// <returns>
+        /// An IDisposable that can be disposed to disable the DependencyCollectorDiagnosticListener.
+        /// </returns>
+        public static IDisposable Enable(TelemetryConfiguration configuration = null)
+        {
+            if (configuration == null)
+            {
+                configuration = TelemetryConfiguration.Active;
+            }
+
+            return DiagnosticListener.AllListeners.Subscribe(new DependencyCollectorDiagnosticListener(configuration));
+        }
+
+        /// <summary>
+        /// Source instrumentation header that is added by an application while making http requests and retrieved by the other application when processing incoming requests.
+        /// </summary>
+        internal const string SourceInstrumentationKeyHeader = "x-ms-request-source-ikey";
+
+        /// <summary>
+        /// Target instrumentation header that is added to the response and retrieved by the calling application when processing incoming responses.
+        /// </summary>
+        internal const string TargetInstrumentationKeyHeader = "x-ms-request-target-ikey";
+
+        /// <summary>
+        /// Standard parent Id header.
+        /// </summary>
+        internal const string StandardParentIdHeader = "x-ms-request-id";
+
+        /// <summary>
+        /// Standard root id header.
+        /// </summary>
+        internal const string StandardRootIdHeader = "x-ms-request-root-id";
+
+        private readonly ApplicationInsightsUrlFilter applicationInsightsUrlFilter;
+        private readonly TelemetryClient client;
+        private readonly ConcurrentDictionary<Guid, DependencyTelemetry> pendingTelemetry = new ConcurrentDictionary<Guid, DependencyTelemetry>();
+
+        internal DependencyCollectorDiagnosticListener(TelemetryConfiguration configuration)
```

Comment:
Is there a way to turn off StyleCop for my unit tests? I don't think its necessary to write XML documentation for each of my methods (your other unit tests certainly don't).